pkg/volume/glusterfs/glusterfs.go

```diff
@@ -967,7 +968,13 @@ func getClusterNodes(cli *gcli.Client, cluster string) (dynamicHostIps []string,
 		if err != nil {
 			return nil, fmt.Errorf("failed to get host ipaddress: %v", err)
 		}
+
 		ipaddr := dstrings.Join(nodeInfo.NodeAddRequest.Hostnames.Storage, "")
+		// This should allow hostname or IP, See #4447.
+		ip := net.ParseIP(ipaddr)
```

Comment:
the comment just above says it can be hostname or IP, yet this line expects only IP. So what is correct here?

Reply:
Yeah, that comment is wrong. Because we can not fill hostname to endpoint address field. It should be a valid IP address.